packages/amplify-graphql-api-construct-tests/src/__tests__/migration/migration-validation.test.ts

```diff
@@ -0,0 +1,179 @@
+import * as path from 'path';
+import { createNewProjectDir, deleteProjectDir, deleteProject } from 'amplify-category-api-e2e-core';
+import { CloudFormationClient, ListStacksCommand, DescribeStackEventsCommand, StackEvent } from '@aws-sdk/client-cloudformation';
+import { initCDKProject, cdkDeploy, cdkDestroy, createGen1ProjectForMigration, deleteDDBTables } from '../../commands';
+import { TestDefinition, writeStackConfig, writeTestDefinitions, writeOverrides } from '../../utils';
+import { DURATION_20_MINUTES } from '../../utils/duration-constants';
+
+jest.setTimeout(DURATION_20_MINUTES);
+
+describe('Migration table import validation', () => {
+  let gen1ProjRoot: string;
+  let gen2ProjRoot: string;
+  let gen1ProjFolderName: string;
+  let gen2ProjFolderName: string;
+  let dataSourceMapping: Record<string, string>;
+
+  beforeAll(async () => {
+    gen1ProjFolderName = 'validategen1';
+    gen1ProjRoot = await createNewProjectDir(gen1ProjFolderName);
+
+    const { DataSourceMappingOutput } = await createGen1ProjectForMigration(
+      gen1ProjFolderName,
+      gen1ProjRoot,
+      'simple_model_public_auth.graphql',
+    );
+    dataSourceMapping = JSON.parse(DataSourceMappingOutput);
+  });
+
+  beforeEach(async () => {
+    gen2ProjFolderName = 'validategen2';
+    gen2ProjRoot = await createNewProjectDir(gen2ProjFolderName);
+    const templatePath = path.resolve(path.join(__dirname, '..', 'backends', 'configurable-stack'));
+    await initCDKProject(gen2ProjRoot, templatePath);
+  });
+
+  afterEach(async () => {
+    try {
+      await cdkDestroy(gen2ProjRoot, '--all');
+    } catch (_) {
+      /* No-op */
+    }
+    deleteProjectDir(gen2ProjRoot);
+  });
+
+  afterAll(async () => {
+    try {
+      await deleteProject(gen1ProjRoot);
+    } catch (_) {
+      /* No-op */
+    }
+
+    try {
+      // Tables are set to retain when migrating from gen 1 to gen 2
+      // delete the tables to prevent resource leak after test is complete
+      await deleteDDBTables(Object.values(dataSourceMapping));
+    } catch (_) {
+      /* No-op */
+    }
+
+    deleteProjectDir(gen1ProjRoot);
+  });
+
+  /*
+  AttributeDefintions - done
+  KeySchema
+  GlobalSecondaryIndexes - done
+  BillingModeSummary - done 
+  ProvisionedThroughput
+  StreamSpecification
+  SSEDescription
+  DeletionProtectionEnabled
+  */
+
+  // Test cases for test.each need to be defined before beforeAll.
+  // The tests cases need data defined in beforeAll.
+  // Define the test name first then get the definition later (after beforeAll).
+  // For each test case, there should be a matching key in the test definitions.
+  // The test definitions include a schema, overrides, and the expected error messages.
+  const testCases = ['extraGSIOnGen2', 'billingMode'];
+  const getTestDefinition = (testCaseName: string): [TestDefinition, string, string[]] => {
```

Comment:
nit for a followup PR: Can we add a typealias for the `TestDefinition, string, string[]` structure to document what data is actually being passed in the test data? Looks to me like it's a test def, some kind of custom override code, and an array of expected errors?